### PR TITLE
Initial work in upgrading to Brave v4.x

### PR DESCRIPTION
**Pull request by jplock**

### CODE REVIEW

File: zipkin-rx/src/main/java/com/smoketurner/dropwizard/zipkin/rx/BraveRxJavaSchedulersHook.java [hunk 1]
```diff
@@ -16,29 +16,27 @@
...
-    public BraveRxJavaSchedulersHook(final Brave brave) {
-        this.brave = Objects.requireNonNull(brave);
+    public BraveRxJavaSchedulersHook(final Tracing tracing) {
+        this.tracing = Objects.requireNonNull(tracing);
     }
 
     @Override
     public Action0 onSchedule(final Action0 action) {
-        final ServerSpanThreadBinder binder = brave.serverSpanThreadBinder();
+        final ServerSpanThreadBinder binder = tracing.serverSpanThreadBinder();
```
Comment (jplock) [inline zipkin-rx/src/main/java/com/smoketurner/dropwizard/zipkin/rx/BraveRxJavaSchedulersHook.java]: @spudone this doesn't currently work.  Any idea how we should support Brave 4.3?

File: zipkin-core/src/main/java/com/smoketurner/dropwizard/zipkin/AbstractZipkinFactory.java [hunk 2]
```diff
@@ -145,27 +148,44 @@ private static int toInt(final String ip) {
...
-        environment.jersey().register(BraveTracingFeature.create(brave));
+        final Tracing tracing = Tracing.newBuilder()
+                .currentTraceContext(MDCCurrentTraceContext.create())
+                .localServiceName(serviceName).localEndpoint(endpoint)
+                .reporter(reporter).sampler(getSampler())
+                .traceId128Bit(traceId128Bit).build();
 
-        return Optional.of(brave);
+        // Register the tracing feature for client and server requests
+        environment.jersey().register(TracingFeature.create(tracing));
```
Comment (adriancole) [inline zipkin-core/src/main/java/com/smoketurner/dropwizard/zipkin/AbstractZipkinFactory.java]: you might want to accept HttpTracing, as this allows you to customize span name and tagging policy https://github.com/openzipkin/brave/tree/master/instrumentation/http#span-data-policy

File: zipkin-rx/src/main/java/com/smoketurner/dropwizard/zipkin/rx/BraveRxJavaSchedulersHook.java (hunk 1 above)
Comment (adriancole) [inline zipkin-rx/src/main/java/com/smoketurner/dropwizard/zipkin/rx/BraveRxJavaSchedulersHook.java]: Tracer.withSpanInScope or CurrentTraceContext should be pointers on this. Let me know if you'd like more

File: zipkin-core/src/main/java/com/smoketurner/dropwizard/zipkin/AbstractZipkinFactory.java [hunk 3]
```diff
@@ -145,27 +148,46 @@ private static int toInt(final String ip) {
...
-                serviceName).reporter(reporter).traceSampler(getSampler())
-                        .traceId128Bit(traceId128Bit).build();
+        final Endpoint endpoint = Endpoint.builder().ipv4(toInt(serviceHost))
+                .port(servicePort).serviceName(serviceName).build();
 
-        // Register the tracing feature for client and server requests
-        environment.jersey().register(BraveTracingFeature.create(brave));
+        final Tracing tracing = Tracing.newBuilder()
+                .currentTraceContext(MDCCurrentTraceContext.create())
+                .localServiceName(serviceName).localEndpoint(endpoint)
```
Comment (adriancole) [inline zipkin-core/src/main/java/com/smoketurner/dropwizard/zipkin/AbstractZipkinFactory.java]: localServiceName is redundant when specifying localEndpoint

**Comment by adriancole** [comment]
Thx!

**Pull request merged**